File: 14-longest-common-prefix/longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }

    prefix := ""
    firstWord := strs[0]

    for i := 0; i < len(firstWord); i++ {
        char := firstWord[i]
        for j := 1; j < len(strs); j++ {
            if i >= len(strs[j]) || strs[j][i] != char {
                return prefix
            }
        }
        prefix += string(char)
    }

    return prefix
}

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) {
            return "";
        }

        String prefix = "";
        String firstWord = strs[0];
        for (int i = 0; i < firstWord.length(); i++) {
            for (String str : strs) {
                if (i >= str.length() || str.charAt(i) !=  firstWord.charAt(i)) {
                    return prefix;
                }
            }
            prefix += firstWord.charAt(i);
        }

        return prefix;
    }
}